src/Microsoft.Azure.WebJobs.Extensions.Kafka/Listeners/KafkaTargetScaler.cs

```diff
@@ -0,0 +1,158 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using Confluent.Kafka;
+using Microsoft.Azure.WebJobs.Host.Scale;
+using Microsoft.Extensions.Logging;
+using System;
+using System.Threading.Tasks;
+
+namespace Microsoft.Azure.WebJobs.Extensions.Kafka
+{
+    public class KafkaTargetScaler<Tkey, TValue> : ITargetScaler
+    {
+        private readonly string topicName;
+        private readonly string consumerGroup;
+        private readonly long lagThreshold;
+        private readonly ILogger logger;
+        private readonly KafkaMetricsProvider<Tkey, TValue> metricsProvider;
+
+        protected DateTime lastScaleUpTime;
+        protected TargetScalerResult lastTargetScalerResult;
+
+        public TargetScalerDescriptor TargetScalerDescriptor { get; }
+ 
+        internal KafkaTargetScaler(string topic, string consumerGroup, string functionID, IConsumer<Tkey, TValue> consumer, KafkaMetricsProvider<Tkey, TValue> metricsProvider, long lagThreshold, ILogger logger)
+        {
+            if (string.IsNullOrWhiteSpace(topic))
+            {
+                throw new ArgumentException("Invalid topic: ", nameof(topic));
+            } 
+
+            if (string.IsNullOrWhiteSpace(consumerGroup))
+            {
+                throw new ArgumentException("Invalid consumer group: ", nameof(consumerGroup));
+            }
+
+            this.topicName = topic;
+            this.consumerGroup = consumerGroup;
+            this.TargetScalerDescriptor = new TargetScalerDescriptor(functionID);
+            this.lagThreshold = lagThreshold;
+            this.logger = logger;
+            this.metricsProvider = metricsProvider;
+
+            this.lastScaleUpTime = DateTime.MinValue;
+            this.lastTargetScalerResult = null;
+
+            this.logger.LogInformation($"Started Target Scaler - topic name: {topicName}, consumerGroup: {consumerGroup}, functionID: {functionID}, lagThreshold: {lagThreshold}.");
+        }
+
+        public async Task<TargetScalerResult> GetScaleResultAsync(TargetScalerContext context)
+        {
+            var metrics = await Task.Run(ValidateAndGetMetrics);
```

Comment:
why just do not call `var metrics = await ValidateAndGetMetrics();`

Reply:
Changed to ` await ValidateAndGetMetrics();`. There was no initial reason.